### PR TITLE
[releaser] many improvements of releaser

### PR DESCRIPTION
**Pull request by TomasLudvik**
| Q             | A
| ------------- | ---
|Description, reason for the PR| We user our internal releaser tool for every release of Shopsys Framework. We had many suggestions of missing instructions, outdated instructions or not working parts. All of them, except changelog generator has been resolved here. Most of the suggestions from #1424 has been resolved here.
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: utils/releaser/src/ReleaseWorker/Release/CreateAndPushGitTagsExceptProjectBaseReleaseWorker.php [hunk 1]
```diff
@@ -58,6 +58,21 @@ public function work(Version $version): void
         $tempDirectory = trim($this->processRunner->run('mktemp -d -t shopsys-release-XXXX'));
         $packageNamesWithProblems = [];
 
+        $this->symfonyStyle->note(sprintf(
+            'In case you do not have saved GIT credentials you may want to cache them temporarily so you do not need to fill them for each repository.'
+            . ' This can be done by using following command `%s`',
+            'git config --global credential.helper "cache --timeout=3600"'
```
[the remark this answers is not in the document]
Comment (TomasLudvik) [reply utils/releaser/src/ReleaseWorker/Release/CreateAndPushGitTagsExceptProjectBaseReleaseWorker.php:64]: It should be sufficient for creation of tags

File: utils/releaser/src/ReleaseWorker/Release/CreateAndPushGitTagsExceptProjectBaseReleaseWorker.php [hunk 2]
```diff
@@ -73,6 +73,12 @@ public function work(Version $version): void
             $this->processRunner->run('git config --global credential.helper "cache --timeout=3600"');
         }
 
+        $this->symfonyStyle->note(
+            'You will be asked for your Github credentials if you have not saved them yet.
+            As we require two factor authentication, you will need to provide repo scope token instead of password.
+            Token can be generated here: https://github.com/settings/tokens/new'
```
[the remark this answers is not in the document]
Comment (TomasLudvik) [reply utils/releaser/src/ReleaseWorker/Release/CreateAndPushGitTagsExceptProjectBaseReleaseWorker.php:79]: It is there on the second line

File: utils/releaser/src/ReleaseWorker/ReleaseCandidate/CheckLatestVersionOfReleaserReleaseWorker.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        return Stage::RELEASE_CANDIDATE;
+    }
+
+    /**
+     * @param \PharIo\Version\Version $version
+     */
+    public function work(Version $version): void
+    {
+        $this->symfonyStyle->note('It is possible, that in current master branch there is improved version of Releaser. Check that and update releaser in version branch if so.');
```
[the remark this answers is not in the document]
Comment (TomasLudvik) [reply utils/releaser/src/ReleaseWorker/ReleaseCandidate/CheckLatestVersionOfReleaserReleaseWorker.php]: I tried... :)

File: utils/releaser/src/ReleaseWorker/ReleaseCandidate/CheckLatestVersionOfReleaserReleaseWorker.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        $this->symfonyStyle->warning('Make sure, that you have checked \Shopsys\Releaser\ReleaseWorker\AbstractShopsysReleaseWorker::EXCLUDED_PACKAGES constant, after updating releaser.');
```
[the remark this answers is not in the document]
Comment (TomasLudvik) [reply utils/releaser/src/ReleaseWorker/ReleaseCandidate/CheckLatestVersionOfReleaserReleaseWorker.php]: I tried... :)

File: utils/releaser/src/ReleaseWorker/CheckCorrectReleaseVersionReleaseWorker.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        return Stage::RELEASE_CANDIDATE;
+    }
+
+    /**
+     * @param \PharIo\Version\Version $version
+     */
+    public function work(Version $version): void
+    {
+        if (!str_starts_with($version->getOriginalString(), 'v')) {
```
[the remark this answers is not in the document]
Comment (TomasLudvik) [reply utils/releaser/src/ReleaseWorker/CheckCorrectReleaseVersionReleaseWorker.php:35]: I have added some more checking

**Review by RostislavKreisinger — APPROVED** [review]

**Review by pk16011990 — APPROVED** [review]

**Pull request merged**